### PR TITLE
make the bounds type open to extension

### PR DESCRIPTION
**Pull request by exaexa**
#14 should be rebased atop of this later (I can do)

### CODE REVIEW

File: Project.toml
Comment (stelmo) [inline Project.toml:1]: can we make 0.7 include the MILP PR too?

File: docs/src/metabolic-modeling.jl [hunk 1]
```diff
@@ -226,12 +226,12 @@ function optimized_vars(cs::C.ConstraintTree, objective::C.LinearValue, optimize
     JuMP.@objective(model, JuMP.MAX_SENSE, C.substitute(objective, x))
     function add_constraint(c::C.Constraint)
         b = c.bound
-        if b isa Float64
-            JuMP.@constraint(model, C.substitute(c.value, x) == b)
-        elseif b isa Tuple{Float64,Float64}
+        if b isa C.EqualTo
+            JuMP.@constraint(model, C.substitute(c.value, x) == b.equal_to)
+        elseif b isa C.Between
```
Comment (stelmo) [inline docs/src/metabolic-modeling.jl:231]: nice

File: docs/src/metabolic-modeling.jl [hunk 2]
```diff
@@ -346,9 +346,9 @@ Dict(k => v.fluxes.R_BIOMASS_Ecoli_core_w_GAM for (k, v) in result.community)
 # both dot-access and array-index syntax.
 
 # You can thus, e.g., set a single bound:
-c.exchanges.oxygen.bound = (-20.0, 20.0)
+c.exchanges.oxygen.bound = C.Between(-20.0, 20.0)
```
Comment (stelmo) [inline docs/src/metabolic-modeling.jl:349]: cool

**Review by stelmo — APPROVED** [review]
this is really great

File: docs/src/metabolic-modeling.jl (hunk 1 above)
Comment (exaexa) [reply docs/src/metabolic-modeling.jl:231]: technically people could even go for something like an overloaded `constrainize(x::Bound)` but let's not go too far at this point...  maybe in the MILP tutorial.

File: Project.toml
Comment (exaexa) [reply Project.toml:1]: yes that's the point. This needs usage docs.

**Pull request merged**